routing_common/transit_types.hpp

```diff
@@ -460,6 +462,19 @@ class Network
   std::string m_title;
 };
 
+template <class Item>
+bool IsValid(std::vector<Item> const & items)
+{
+  return std::all_of(items.cbegin(), items.cend(), [](Item const & item) { return item.IsValid(); });
+}
+
+template <class Item>
+bool IsValidSortedUnique(std::vector<Item> const & items)
+{
+  return IsValid(items) && std::is_sorted(items.cbegin(), items.cend()) &&
+         std::adjacent_find(items.cbegin(), items.cend()) == items.cend();
```

Comment:
We have IsSortedAndUnique() in base/stl_add.hpp.

Reply:
Done